arwen/host/execution.go

```diff
@@ -254,6 +254,23 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 	scExecutionInput := input
 
 	blockchain := host.Blockchain()
+
+	isSmartContract := blockchain.IsSmartContract(input.RecipientAddr)
```

Comment:
I think this code should be more inside exeucteOnDestContextNoBuiltInFunction.

Reply:
Wouldn't it be better to have the code deep inside func (host *vmHost) execute(input *vmcommon.ContractCallInput) error ?

Reply:
refactored and used a logic similar with the one from builtin functions